clang/test/Sema/debug-93066.cpp

```diff
@@ -0,0 +1,14 @@
+// RUN: %clang_cc1 -std=c++23 -fsyntax-only -verify %s
+
```

Comment:
Judging by these here, this is going in the right direction, though I would also want to see some more tests for other expressions that are not themselves modifiable lvalues, and also in other contexts that require modifiable lvalues, e.g.
```c++
reinterpret_cast<int*>(42) += 3;
```
There are some more in the description of the issue iirc, but feel also free to think of some more cases to test.

Reply:
Sure, I'll add more.